pkg/master/controller.go

```diff
@@ -257,18 +260,20 @@ type EndpointReconciler interface {
 // masterCountEndpointReconciler reconciles endpoints based on a specified expected number of
 // masters. masterCountEndpointReconciler implements EndpointReconciler.
 type masterCountEndpointReconciler struct {
-	masterCount      int
-	endpointRegistry endpoint.Registry
+	masterCount       int
+	endpointRegistry  endpoint.Registry
+	configmapRegistry configmap.Registry
```

Comment:
I'm not sure I follow.

Just to clarify, when you say "Kubelet reads the configmap", specifically which configmap are you referring to?

Are you saying that the Kubelet reads API server config from the API server, then saves that to a file on the corresponding node's disk, then the API server watches that file for changes via inotify (I guess you can inotify over a network?), then decides whether to restart if it changes?